src/util/psbt/map/input.rs

```diff
@@ -155,6 +157,38 @@ pub struct PsbtSigHashType {
     pub (in ::util::psbt) inner: u32,
 }
 
+impl fmt::Display for PsbtSigHashType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self.ecdsa_hash_ty() {
```

Comment:
I think this requires it's own custom code as this is the unification of both sighash types. You can write FromStr accordingly. We should **not** serialize `RESERVED` (as there is no such thing). And error on things that are not strings

```rust
impl fmt::Display for PsbtSigHashType {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        let s = match self.as_u32() {
            0 => "SIGHASH_DEFAULT"
            1 => "SIGHASH_ALL",
            .. => "SIGHASH_NONE",
            .. => "SIGHASH_SINGLE",
            ..=> "SIGHASH_ALL|SIGHASH_ANYONECANPAY",
            .. => "SIGHASH_NONE|SIGHASH_ANYONECANPAY",
            .....
        };
        f.write_str(s)
    }
}

```